### PR TITLE
Enable src-alias for keystore

### LINKED ISSUE #462: Keystores with more than one cert
I am using the approach of self provided key- and truststores and have a keystore that includes more than one certificate. For this scenario the ssl-role extracts like this:

```yml
    - name: Convert Keystore to Pem Format
      shell: |
        keytool -noprompt -importkeystore \
          -srckeystore {{keystore_path}} \
          -srcstorepass {{keystore_storepass}} \
          -destkeystore /var/ssl/private/generation/{{service_name}}.p12 \
          -deststoretype PKCS12 \
          -deststorepass {{keystore_storepass}} \
          -destkeypass {{keystore_storepass}} {{extra_args}}
```

However this fails, since the keystore contains more than one certificate and does not know which to use. My fix is currently to add an additional parameter `-srcalias {{inventory_hostname}}` to the keytool command.

I did not yet get to test, whether that would work also for keystores that contain only one certificate, but I assume it might. Do you see this as a possible improvement for upstream? :)

### PR DESCRIPTION
**Pull request by Fobhep**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #462 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: VARIABLES.md [hunk 1]
```diff
@@ -180,6 +180,13 @@ Default:  false
 
 ***
 
+### ssl_keystore_alias
+
+Keystore source alias for host specific certificate if keystore contains more than one certificate. Used with ssl_provided_keystore_and_truststore: true. May set per host, or use inventory_hostname variable eg "{{inventory_hostname}}"
```
Comment (JumaX) [inline VARIABLES.md:185]:
```suggestion
The alias for a host specific certificate, when the keystore contains more than one certificate. Used with ssl_provided_keystore_and_truststore: true. Can be set per host, or use inventory_hostname variable eg "{{inventory_hostname}}"
```

File: hosts_example.yml [hunk 2]
```diff
@@ -61,6 +61,7 @@ all:
     # ssl_keystore_filepath: "/tmp/certs/{{inventory_hostname}}-keystore.jks"
     # ssl_keystore_key_password: mystorepassword
     # ssl_keystore_store_password: mystorepassword
+    # ssl_keystore_alias: {{ inventory_hostname }} # if your keystore contains more than one cert, you can specify one here
```
Comment (JumaX) [inline hosts_example.yml:64]:
```suggestion
    # ssl_keystore_alias: {{ inventory_hostname }}
```

**Comment by JumaX** [comment]
@Fobhep Please also fill in the details on how this has been tested.

File: hosts_example.yml [hunk 3]
```diff
@@ -61,6 +61,7 @@ all:
     # ssl_keystore_filepath: "/tmp/certs/{{inventory_hostname}}-keystore.jks"
     # ssl_keystore_key_password: mystorepassword
     # ssl_keystore_store_password: mystorepassword
+    # ssl_keystore_alias: {{ inventory_hostname }}
```
Comment (domenicbove) [inline hosts_example.yml:64]:
```suggestion
    # ssl_keystore_alias: <alias for host specific certificate, only required if multiple certs in provided keystore>
```

File: VARIABLES.md [hunk 4]
```diff
@@ -180,6 +180,13 @@ Default:  false
 
 ***
 
+### ssl_keystore_alias
```
[the remark this answers is not in the document]
Comment (domenicbove) [reply VARIABLES.md:183]: Then again I'm worried that if we default the variable to the empty string then the if statement in the code might think its defined, this should be checked

**Review by domenicbove — CHANGES_REQUESTED** [review]
Great start! A few documentation things. Also this PR should be targeting the 6.1.x branch instead of master

File: VARIABLES.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Fobhep) [reply VARIABLES.md:183]: ooh - sorry. I totally missed that that was dynamic :D I will adjust that and the jinja dynamic if necessary.

**Comment by Fobhep** [comment]
> Great start! A few documentation things. Also this PR should be targeting the 6.1.x branch instead of master

thx for the quick review - will fix those things :)

**Comment by Fobhep** [comment]
@domenicbove I set the base branch now to 6.1.x and rebased also to 6.1.x

However if you want the Jolokia TLS settings/vars not merged here - I can either wait for it to be merged elsewhere, or rebase to some other branch - WDYT?

Regarding testing @JumaX - I used this "fix" succesfully in an environment where we had a keystore with multiple certs. Most clean way would probably however be to write a molecule scenario - similar to your one with provided key/truststore, which I currently sadly do not have the time for.

File: VARIABLES.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Fobhep) [reply VARIABLES.md]: I used now  `src_keystore_alias|length` - which should only trigger if the string is > 0.
Do you think that is sufficient? Alternatively one could do a 
`src_keystore_alias.strip()|length` to ensure somebody is not accidentally putting an empty space into the var - but that might crash with an actual existing srcalias containing spaces, right?

File: hosts_example.yml [hunk 5]
```diff
@@ -61,9 +61,9 @@ all:
     # ssl_keystore_filepath: "/tmp/certs/{{inventory_hostname}}-keystore.jks"
     # ssl_keystore_key_password: mystorepassword
     # ssl_keystore_store_password: mystorepassword
+    # ssl_keystore_alias: <alias for host specific certificate, only required if multiple certs in provided keystore>
     # ssl_truststore_filepath: "/tmp/certs/truststore.jks"
     # ssl_truststore_password: truststorepass
-    # ssl_truststore_ca_cert_alias: <alias to the ca certificate in the truststore eg. CARoot>
```
Comment (domenicbove) [inline hosts_example.yml:66]: this line should stay i think

File: roles/confluent.ssl/tasks/provided_keystore_and_truststore.yml [hunk 6]
```diff
@@ -46,7 +46,7 @@
       shell: |
         keytool -noprompt -importkeystore \
           -srckeystore {{keystore_path}} \
-          -srcstorepass {{keystore_storepass}} \
+          -srcstorepass {{keystore_storepass}}{% if ssl_keystore_alias|length %} -srcalias {{ssl_keystore_alias}} {% endif %} \
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/provided_keystore_and_truststore.yml:49]: if ssl_keystore_alias|length returns 0 then the if statement fails? might be more clear to just say ssl_keystore_alias|length == 0

File: hosts_example.yml (hunk 5 above)
Comment (Fobhep) [reply hosts_example.yml:66]: of course - fixed

File: roles/confluent.ssl/tasks/provided_keystore_and_truststore.yml (hunk 6 above)
Comment (Fobhep) [reply roles/confluent.ssl/tasks/provided_keystore_and_truststore.yml:49]: exactly - my writing is the abbreviated version - but I like your approach to make it easier to read

File: roles/confluent.variables/defaults/main.yml [hunk 7]
```diff
@@ -112,6 +112,9 @@ ssl_keystore_key_password: ""
 ### Keystore Password for host specific keystore. Used with ssl_provided_keystore_and_truststore: true. May set per host if keystores have unique passwords
 ssl_keystore_store_password: ""
 
+### Keystore source alias for host specific certificate if keystore contains more than one certificate. Used with ssl_provided_keystore_and_truststore: true. May set per host, or use inventory_hostname variable eg "{{inventory_hostname}}"
```
Comment (domenicbove) [inline roles/confluent.variables/defaults/main.yml:115]:
```suggestion
### Keystore source alias for host specific certificate. Only required if keystore contains more than one certificate. Used with ssl_provided_keystore_and_truststore: true. May set per host, or use inventory_hostname variable eg "{{inventory_hostname}}"
```

I want to make this more clear it is not a required variable

Comment (Fobhep) [reply roles/confluent.variables/defaults/main.yml]: agreed!

File: roles/confluent.ssl/tasks/provided_keystore_and_truststore.yml (hunk 6 above)
Comment (Fobhep) [reply roles/confluent.ssl/tasks/provided_keystore_and_truststore.yml:49]: I will rewrite it with `if ssl_keystore_alias|length > 0`

**Review by domenicbove — APPROVED** [review]
Great work. so fyi this 61x branch will have all the new features to be released w 610. But it still installs 600

**Comment by JumaX** [comment]
@Fobhep Great job!  We've been discussing this internally, we would like for there to be a test in place in one of the existing molecule scenarios validate this works as expected.  It would involve updating the certs generation script for the given scenario to add another certificate with a unique alias.  We could then update a tls molecule scenario to check the contents of the keystore for the cert with the additional alias.  We're trying to make it mandatory (for ourselves as well) to add test for each piece of new functionality, as we've been bitten pretty hard in the past by missing out on such checks.  I hope this makes sense, and if you have any questions about creating the test please don't hesitate to let us know.

**Comment by Fobhep** [comment]
@JumaX @domenicbove  Absolutely I think tests always make more sense than not having any tests ;)

What do you think would be the best way to get the additional certificate into the provided keystore? Should we adapt the bash script that generates all key and trust stores in general, or should we create a custom one that is only available in the space of the custom scenario?

**Comment by domenicbove** [comment]
Hi @Fobhep I create this pr into your branch: which should have an automated test: https://github.com/ATIX-AG/cp-ansible/pull/3

It also merged in the latest 6.1.x. Please review and merge and then this PR should be good to go!

**Comment by Fobhep** [comment]
@domenicbove oh wow - thanks for the effort! I merged it already into my branch but I will give it another look and then let you know :)

**Comment by Fobhep** [comment]
@domenicbove ok, looks good to me :+1: thanks again :)

File: roles/confluent.test/molecule/certs-create.sh [hunk 8]
```diff
@@ -62,26 +62,40 @@ for line in `sed '/^$/d' $filename`; do
...
+      FORMAT=$1
+
+      source=""
+      samples=("")
+      if [ $1 = "country" ]; then
+         source="country"
+         samples="US Canada Mexico..."
+      else
+        echo "try again"
+      fi
```
Comment (domenicbove) [inline roles/confluent.test/molecule/certs-create.sh:74]:
```suggestion
```

whoops that should not have made it into the script, that was some sample code i found off the internet

**Review by domenicbove — APPROVED** [review]
Looks good, minus one little change

File: roles/confluent.test/molecule/certs-create.sh (hunk 8 above)
Comment (Fobhep) [reply roles/confluent.test/molecule/certs-create.sh:74]: Ah ok - I will remove it then :)

**Pull request merged**